r/R/util.R

```diff
@@ -58,10 +58,22 @@ r_symbolic_constants <- c(
   "NA_integer_", "NA_real_", "NA_complex_", "NA_character_"
 )
 
+is_function <- function(expr, name) {
+  if (!is.call(expr)) {
+    return(FALSE)
+  } else {
+    if (deparse(expr[[1]]) == name) {
+      return(TRUE)
+    }
+    out <- lapply(expr, is_function, name)
+  }
+  any(vapply(out, isTRUE, TRUE))
+}
+
 all_funs <- function(expr) {
   # Don't use setdiff so that we preserve duplicates
```

Comment:
```suggestion
  # Don't just setdiff(all.names, all.vars) because that would drop functions that share a name with a var
```

Reply:
IIRC there was some other reason that we didn't want to dedupe the names in this

Reply:
Oh, hah, I see—you changed this in https://github.com/apache/arrow/pull/11108/files#diff-f6235d4767fc4a7ee1bb726d816b9742ef0bc07503dceb678fd3bc55ee15454bR62

Reply:
@nealrichardson I'll defer to you on the comment here because I don't fully follow the logic behind why this was removed in #11108

Reply:
I changed it because it seemed overly complicated for something that base R should do for us, and no test failed when I did.

Reply:
For the comment, maybe something like this?
>It is not sufficient to simply do something like `setdiff(all.names, all.vars)` here
>because that would fail to return the names of functions that share names with
>variables. To preserve duplicates, call `all.names()` not `all_names()` here.